src/lidar/read_rp_lidar.cpp

```diff
@@ -82,9 +70,8 @@ bool initializeLidar() {
 DataPoint<points_t> readLidar() {
     if (lidar_initialized) {
         DataPoint<points_t> pts;
-        points_lock.lock();
+        std::lock_guard<std::mutex> lk(points_lock);
         pts = last_points;
-        points_lock.unlock();
         return pts;
     }
     return {};
```

Comment:
`lidar_initialized` is protected by `points_lock` so the `lock_guard` should be created *before* the if statement.